### PR TITLE
The One With Hooks

### PR DESCRIPTION
**Pull request by develohpanda**
[We finally did it.](https://www.youtube.com/watch?v=dQw4w9WgXcQ) 

> For reference, the original error was that insomnia-components was unable to locate the react-dom module. Changing the way we alias resolves that issue.

This error would surface as the common invalid react hook error seen [here](https://reactjs.org/warnings/invalid-hook-call-warning.html).

### CODE REVIEW

File: packages/insomnia-app/package.json [hunk 1]
```diff
@@ -74,7 +74,6 @@
     "nunjucks",
     "pdfjs-dist",
     "prop-types",
-    "react",
```
Comment (develohpanda) [inline packages/insomnia-app/package.json]: This was the culprit

File: packages/insomnia-components/webpack.config.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     library: 'insomniaComponents',
     libraryTarget: 'commonjs2',
   },
+  externals: ['react'],
```
Comment (develohpanda) [inline packages/insomnia-components/webpack.config.js]: Tell `webpack` to not bundle react, when packing `insomnia-components`

**Review by gschier — COMMENTED** [review]
I _think_ setting React as an `external` in Webpack (by removing it from `packedDependencies`) means that we'd be using the development version in production (since it won't be processed) which could have a significant performance hit.

It would be nice to figure out the root of the problem.

File: packages/insomnia-components/webpack.config.js (hunk 2 above)
Comment (gschier) [reply packages/insomnia-components/webpack.config.js]: Should also have `react-dom` in here

**Comment by develohpanda** [comment]
My understanding is setting mode to production achieves that result, not externals. See the last paragraph [here](https://webpack.js.org/guides/production/#specify-the-mode).

But you're right, setting as external in the app doesn't seem correct.

**Comment by gschier** [comment]
Production mode is definitely required, but I don't think it will do anything to externals since externals are not included in the outputted bundle. Specifying `module-name` as external causes webpack to not process it at all and will simply include `require('module-name')` in the bundle instead.

On a positive note, I stumbled upon a solution to the `react` external testing out this fix. See #2292 which uses this branch as a base.

**Review by gschier — APPROVED** [review]
Yay, it feels so good to have gotten to the bottom of this

**Pull request merged**